File: src/main/resources/static/scripts/controller/blogueController.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


angular.module('aplikasiBlogue')
        .controller('blogueController',function ($scope,blogueService){
            $scope.hallo = "Hello";
            $scope.nama = "Blogue";
            $scope.model = {};
            $scope.isEdit = false;
            
            $scope.tampilkanData = function() {
                blogueService.getListBlogue().success(function (data){
                    $scope.dataServer = data;
                })
            };
            
            $scope.data = [
                {"nama": "Blogue","alamat": "Jakarta"},{"nama": "Go Blog","alamat": "Jakarta"}
            ];
            
            $scope.hapus = function (index){
                $scope.data.splice(index,1);
            };
            
            $scope.simpan = function(data){
                if ($scope.isEdit==true){
                    $scope.data[$scope.indexEdit] = data;
                }
                else{
                    $scope.data.push(data);
                }
                $scope.clear();
            }
            
            $scope.clear = function () {
                $scope.model = {};
                $scope.isEdit = false;
            }
            
            $scope.edit = function (data,index) {
                $scope.model.nama = data.nama;
                $scope.model.alamat = data.alamat;
                $scope.indexEdit = index;
                $scope.isEdit = true;
            }
            
            $scope.tampilkanData = function () {
                blogueService.getBlogueFromDb().success(function (data){
                   $scope.dataServer = data.content; 
                });
            };
            
            $scope.isDataEdit = false;
            $scope.currentBlogue = {};
            $scope.simpanData = function (data){
                if ($scope.isDataEdit==true){
                    blogueService.update(data.id,data).success(function (data){
                       alert("data berhasil dirubah")
                       $scope.clearData();
                    }).error(function (){
                        alert("terjadi kesalahan dalam penyimpanan data")
                    })
                }
                else{
                    blogueService.save(data).success(function (data){
                        alert("data berhasil disimpan")
                        $scope.clearData();
                    }).error(function (){
                        alert("Terjadi kesalahan dalam menyimpan data")
                    })
                }
            }
            
            $scope.hapusData = function(id) {
                blogueService.delete(id).success(function (data){
                    alert("data berhasil dihapus");
                    $scope.clearData();
                }).error(function (){
                    alert("terjadi kesalahan dalam penghapusan data");
                })
            }
            
            $scope.editData = function (data){
                $scope.isDataEdit = true;
                $scope.currentBlogue = {};
                $scope.currentBlogue.id = data.id;
                $scope.currentBlogue.tanggal = data.tanggal;
                $scope.currentBlogue.judul = data.judul;
                $scope.currentBlogue.isi = data.isi;
                $scope.currentBlogue.author = data.author;
                
            };
            
            $scope.clearData = function (){
                $scope.isDataEdit = false;
                $scope.currentBlogue = {};
                $scope.reloadData();
            }
            
            $scope.reloadData = function (){
                blogueService.getBlogueFromDb().success(function (data){
                    $scope.dataServer = data.content;
                })
            }
});

File: src/main/java/blogue/springboot/controller/BlogueController.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package blogue.springboot.controller;

import blogue.springboot.domain.Blogue;
import blogue.springboot.service.BlogueDao;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
/**
 *
 * @author DyoRizqal
 */
@RestController
@RequestMapping("/api/blogue")
public class BlogueController {
    @Autowired
    private BlogueDao blogueDao;
    
    @RequestMapping(value = "/{id}",method = RequestMethod.GET)
    public Blogue findBlogueById(@PathVariable String id)
    {
        return blogueDao.findOne(id);
    }
    
    @RequestMapping(method = RequestMethod.POST)
    public void save(@RequestBody Blogue c)
    {
        blogueDao.save(c);
    }
    
    @RequestMapping(method = RequestMethod.GET)
    public Page<Blogue> findAll(Pageable pageable)
    {
        return blogueDao.findAll(pageable);
    }
    
    @RequestMapping(value = "/{id}",method = RequestMethod.DELETE)
    public void delete(@PathVariable String id)
    {
        Blogue c = blogueDao.findOne(id);
        if (c != null) {
            blogueDao.delete(c);
        }
    }
    
    @RequestMapping(value = "/{id}",method = RequestMethod.PUT)
    public void edit(@PathVariable String id, @RequestBody Blogue c)
    {
        Blogue blogue = blogueDao.findOne(id);
        if (blogue != null) {
            c.setId(id);
            blogueDao.save(c);
        }
    }
}
